gui-libs/xdg-desktop-portal-hyprland/xdg-desktop-portal-hyprland-9999.ebuild

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 Julien Roy <julien@jroy.ca>
+# Distributed under the terms of the ISC License
+
+EAPI=7
+
+inherit meson
+
+DESCRIPTION="xdg-desktop-portal backend for hyprland"
+HOMEPAGE="https://github.com/hyprwm/xdg-desktop-portal-hyprland"
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://github.com/hyprwm/${PN}.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/hyprwm/${PN}/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm64 ~ppc64 ~riscv ~x86"
+fi
+
+LICENSE="BSD"
+SLOT="0"
+IUSE="elogind systemd"
+REQUIRED_USE="?? ( elogind systemd )"
+
+DEPEND="
+	>=media-video/pipewire-0.3.41:=
+	dev-libs/inih
+	dev-libs/wayland
+	media-libs/mesa
+	x11-libs/libdrm
+	|| (
+		systemd? ( >=sys-apps/systemd-237 )
+		elogind? ( >=sys-auth/elogind-237 )
+		sys-libs/basu
+	)
+	dev-qt/qtbase
+	dev-qt/qtwayland:6
+"
+# mesa is needed for gbm dep (which it hards sets to 'on')
+RDEPEND="
+	${DEPEND}
+	sys-apps/xdg-desktop-portal
+"
+BDEPEND="
+	>=dev-libs/wayland-protocols-1.24
+	virtual/pkgconfig
+"
+
+src_configure() {
+	local emesonargs=()
+
+	if use systemd; then
+		emesonargs+=(-Dsd-bus-provider=libsystemd)
+	elif use elogind; then
+		emesonargs+=(-Dsd-bus-provider=libelogind)
+	else
+		emesonargs+=(-Dsd-bus-provider=basu)
+	fi
+	meson_src_configure
+}
+
+src_compile() {
+	meson_src_compile
+	emake -C hyprland-share-picker all
```

Comment:
Now that I'm looking closer at this, the build system is a mix of cmake, make and ninja. Dunno what that is trying to do, but could you try to call cmake in the ebuild using cmake eclass, instead of calling cmake through make - https://github.com/hyprwm/xdg-desktop-portal-hyprland/blob/master/hyprland-share-picker/Makefile#L4

Reply:
The only way I found to make it work with cmake is this way:
```
diff --git a/gui-libs/xdg-desktop-portal-hyprland/xdg-desktop-portal-hyprland-9999.ebuild b/gui-libs/xdg-desktop-portal-hyprland/xdg-desktop-portal-hyprland-9999.ebuild
index c1a1102..d1fbc27 100644
--- a/gui-libs/xdg-desktop-portal-hyprland/xdg-desktop-portal-hyprland-9999.ebuild
+++ b/gui-libs/xdg-desktop-portal-hyprland/xdg-desktop-portal-hyprland-9999.ebuild
@@ -45,6 +45,11 @@ BDEPEND="
 	virtual/pkgconfig
 "
 
+src_prepare() {
+	cmake_run_in "${S}/hyprland-share-picker" cmake_src_prepare
+	default
+}
+
 src_configure() {
 	local emesonargs=()
 
@@ -55,15 +60,24 @@ src_configure() {
 	else
 		emesonargs+=(-Dsd-bus-provider=basu)
 	fi
-	meson_src_configure
+	BUILD_DIR="${WORKDIR}/${P}-build" \
+			 EMESON_SOURCE="${WORKDIR}/${P}" \
+			 meson_src_configure
+
+	local mycmakeargs=(
+		-DCMAKE_BUILD_TYPE:STRING=Release
+	)
+	cmake_src_configure
 }
 
 src_compile() {
-	meson_src_compile
-	emake -C hyprland-share-picker all
+	BUILD_DIR="${WORKDIR}/${P}-build" \
+			 EMESON_SOURCE="${WORKDIR}/${P}" \
+			 meson_src_compile
+	cmake_src_compile
 }
 
 src_install() {
-	meson_src_install
-	dobin "${S}/hyprland-share-picker/build/hyprland-share-picker"
+	BUILD_DIR="${WORKDIR}/${P}-build" meson_src_install
+	cmake_src_install
 }
```

I don't think that's the good way, but I couldn't make it work any other way. Having two src_configure seems to make cmake and meson overwrite each other depending on the other they're called. I guess a cleaner solution would be to make a separate ebuild just for hyprland-share-picker, but I'm not sure if that makes sense either since neither package make sense without the other, and they're in the same source code.

Reply:
eek, that's horrible. ok I'll merge the current one, fingers crossed someone doesn't break something. upstream really should try to fix their build system to use a coherent one.